### PR TITLE
Upgrade to Angular 18

### LINKED ISSUE #26208: Upgrade to Angular 18
##### **Overview of the issue**

Angular 18 will be [released this week](https://angular.io/guide/releases#support-policy-and-schedule).

- [x] @angular-eslint
- [x] @angular-architects
- [ ] @angular-builders
- [x] @fortawesome/angular-fontawesome
- [x] @ng-bootstrap
- [x] jest-preset-angular
- [x] ngx-cookie-service
- [x] ngx-infinite-scroll

##### **Motivation for or Use Case**

We should stay up-to-date with major releases of the frameworks we depend on.

### PR DESCRIPTION
**Pull request by mraible**
Fixes #26208. 
Related https://github.com/jhipster/generator-jhipster/pull/26321.

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/angular/resources/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "dependencies": {
-    "@angular/common": "17.3.9",
+    "@angular/common": "18.0.0",
```
Comment (mraible) [inline generators/angular/resources/package.json:3]:
```suggestion
    "@angular/common": "18.0.1",
```

File: generators/angular/resources/package.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
     "@angular-builders/custom-webpack": "17.0.2",
     "@angular-builders/jest": "17.0.3",
     "@angular-eslint/eslint-plugin": "17.4.1",
-    "@angular/cli": "17.3.7",
+    "@angular/cli": "18.0.0",
```
Comment (mraible) [inline generators/angular/resources/package.json:27]:
```suggestion
    "@angular/cli": "18.0.2",
```

**Comment by mraible** [comment]
This won't work until ng-bootstrap has a 17.0.0 release. https://github.com/ng-bootstrap/ng-bootstrap/releases

File: generators/angular/templates/package.json.ejs [hunk 3]
```diff
@@ -107,6 +107,16 @@
...
+    "@angular/compiler-cli": "<%= nodeDependencies['@angular/common'] %>",
+    "@angular/common": "<%= nodeDependencies['@angular/common'] %>",
+    "@angular/compiler": "<%= nodeDependencies['@angular/common'] %>",
+    "@angular/core": "<%= nodeDependencies['@angular/common'] %>",
+    "@angular/forms": "<%= nodeDependencies['@angular/common'] %>",
+    "@angular/localize": "<%= nodeDependencies['@angular/common'] %>",
+    "@angular/platform-browser": "<%= nodeDependencies['@angular/common'] %>",
+    "@angular/platform-browser-dynamic": "<%= nodeDependencies['@angular/common'] %>",
+    "@angular/router": "<%= nodeDependencies['@angular/common'] %>",
+    "@angular-devkit/build-angular": "<%= nodeDependencies['@angular/cli'] %>",
```
Comment (mshima) [inline generators/angular/templates/package.json.ejs:119]:
```suggestion
```

**Review by DanielFran — APPROVED** [review]

**Pull request merged**